src/Microsoft.Docs.Template/Models/Attributes.cs

```diff
@@ -36,21 +34,4 @@ public class HtmlAttribute : DataTypeAttribute { }
     public class UidAttribute : DataTypeAttribute { }
 
     public class XrefAttribute : DataTypeAttribute { }
-
-    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
-    public class LocaleAttribute : ValidationAttribute
-    {
-        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
-        {
-            try
-            {
-                var culture = new CultureInfo(value.ToString());
-                return null;
-            }
-            catch (CultureNotFoundException)
-            {
-                return new ValidationResult($"Locale '{value.ToString()}' is not supported.", null);
```

Comment:
>ValidationResult [](start = 27, length = 16)

why not just throw DocfxException here?

Reply:
+1

---
In reply to: [219429736](https://github.com/dotnet/docfx/pull/3444#discussion_r219429736) [](ancestors = 219429736)

Reply:
Data annotation is primarily used for SDP as an extension mechanism where all the logic is self-contained inside the implementation of a particular attribute.